### PR TITLE
Aryan modal endpoints

### PR DESCRIPTION
**Pull request by aryanbisht-dev**
Added some props from App.js, resolved conflicts from modal.js

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -17,6 +17,8 @@ function App() {
   const vidTimeControls = 5;
   const [recordState, setRecordState] = useState(false);
   const [modalIsOpen, setModalIsOpen] = useState(false);
+  const defaultPlaybackRate = 2.0
+  const defaultVidSource = "https://commondatastorage.googleapis.com/gtv-videos-bucket/sample/Sintel.mp4"
```
Comment (AntiAnt) [inline src/App.js]: this should probably be state in case we want to update the source at any point

Comment (AntiAnt) [reply src/App.js]: this includes the playback rate

File: src/App.js [hunk 2]
```diff
@@ -65,6 +67,7 @@ function App() {
   };
 
   useEffect(() => {
+    video.current.playbackRate = defaultPlaybackRate;
```
Comment (AntiAnt) [inline src/App.js]: how do we know video is not null here?
should proabably move this inside the conditional block

File: src/App.js [hunk 3]
```diff
@@ -130,10 +133,9 @@ function App() {
     video.current.currentTime = startTime;
   }
 
-  // const handleTimer = () => {
-  //   setPlayState(!playState);
-  // };
-
+  function playTime() {
```
Comment (AntiAnt) [inline src/App.js]: play time is a value not a function we need to compute the correct timme and pass it as a number

**Review by AntiAnt — CHANGES_REQUESTED** [review]
take a look at what is happening inside VidModal now. that might help determine what to pass in

File: src/App.js [hunk 4]
```diff
@@ -222,6 +226,8 @@ function App() {
         isOpen={modalIsOpen}
         onClose={setModalIsOpenToFalse}
         onKeyDown={HandleKeyDown}
+        playTime={playTime}
```
Comment (AntiAnt) [inline src/App.js]: So we are still passing the function to VidModal. playTime may not be the best name to represent its value but it represents the currentTime for the video in VidModal to be set to when rendered.
```
const [state, setState] = useState()
...

const handleModalOpen = () => {
    //set isopen true
    // calculate & set modal time
}
...
playTime={state}// this is the calculated time in seconds like: 23.1234
...
```

Take a look how VidModal handles playTime

**Review by AntiAnt — CHANGES_REQUESTED** [review]

File: src/App.js (hunk 4 above)
Comment (aryanbisht-dev) [reply src/App.js]: Would getPlayTime be a  better name?

File: src/App.js [hunk 5]
```diff
@@ -17,13 +17,27 @@ function App() {
   const vidTimeControls = 5;
```
Comment (AntiAnt) [inline src/App.js:17]: are vidTimeControls and SETBACK_CURRENT_TIME functional the same? if so we can combine them

File: src/App.js [hunk 6]
```diff
@@ -17,13 +17,27 @@ function App() {
...
   const [recordState, setRecordState] = useState(false);
   const [modalIsOpen, setModalIsOpen] = useState(false);
+  const [defaultPlaybackRate, setDefaultPlaybackRate] = useState(2.0);
+  const [defaultVidSource, setDefaultVidSource] = useState("https://commondatastorage.googleapis.com/gtv-videos-bucket/sample/Sintel.mp4");
+  const [setBackTime, setSetBackTime] = useState(0)
+  const SETBACK_CURRENT_TIME = 5;
 
   const setModalIsOpenToTrue = () => {
     setModalIsOpen(true);
+    if (video.current.currentTime < 5){
```
Comment (AntiAnt) [inline src/App.js]: we can probably parameterize the 5 here to keep it up to date with any changes to that increment.

File: src/App.js [hunk 7]
```diff
@@ -17,13 +17,27 @@ function App() {
...
+      const setBackTimeStamp = video.current.currentTime - SETBACK_CURRENT_TIME
+      setSetBackTime(setBackTimeStamp)
+    }
+      video.current.pause()
   };
 
   const setModalIsOpenToFalse = () => {
     setModalIsOpen(false);
+    setPlayState(true)
+    video.current.play()
```
Comment (AntiAnt) [inline src/App.js:40]: I want to bring up a point here, we are assuming `video.current.play()` is a synchronous and is a void function. The function returns a Promise making it async. If you have any code that relies on the play function being synchronous you might run into some problems. 

This fine and probably doesn't need any changes jsut something to think about. Here is the [documentation](https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement/play) on the function.

File: src/App.js [hunk 8]
```diff
@@ -130,9 +145,9 @@ function App() {
     video.current.currentTime = startTime;
   }
 
-  // const handleTimer = () => {
-  //   setPlayState(!playState);
-  // };
+  const playTime = () => {
```
Comment (AntiAnt) [inline src/App.js]: We can probably remove playTime()  if its not being used anymore

**Review by AntiAnt — CHANGES_REQUESTED** [review]

**Review by AntiAnt — APPROVED** [review]

**Pull request merged**